### PR TITLE
Show better event URLs

### PR DESCRIPTION
**Pull request by wpf500**
Add a nice slug to the event URLs so they look nicer and have better SEO.

e.g.
`/event/15327407689` becomes `/event/guardian-live-guantanamo-diary-15327407689`

URLs without slugs or with invalid slugs (e.g. after an event name change) redirect to the correct URL.

### CODE REVIEW

File: frontend/app/controllers/Event.scala [hunk 1]
```diff
@@ -38,10 +40,19 @@ trait Event extends Controller {
   private def BuyAction(id: String) = NoCacheAction andThen recordBuyIntention(id) andThen
     authenticated(onUnauthenticated = notYetAMemberOn(_)) andThen memberRefiner()
 
-  def details(id: String) = CachedAction { implicit request =>
-    EventbriteService.getEvent(id).map { event =>
-      eventDetail(event, request.path)
-    }.getOrElse(Redirect(Config.guardianMembershipUrl))
+  def details(slug: String) = NoCacheAction { implicit request =>
+    val eventOpt = for {
+      id <- EBEvent.slugToId(slug)
```
Comment (mattandrews) [inline frontend/app/controllers/Event.scala]: This is more of a learning question than a review, but does redefining `id` here only apply to the scope of the `for`?

Comment (mattandrews) [inline frontend/app/controllers/Event.scala]: ... I can't read.
:eyeglasses:

**Comment by mattandrews** [comment]
:+1: i love urls

**Comment by jamesoram** [comment]
:+1:

**Pull request merged**